ceph_deploy/hosts/centos/install.py

```diff
@@ -157,6 +162,11 @@ def mirror_install(distro, repo_url, gpg_url, adjust_repos, extra_installs=True)
 
 
 def repo_install(distro, reponame, baseurl, gpgkey, **kw):
+    # do we have specific components to install?
+    # removed them from `kw` so that we don't mess with other defaults
+    # note: when split packages for ceph land for CentOS, `packages`
+    # can be used. Unused for now.
+    packages = kw.pop('components', default_components) or default_components  # noqa
```

Comment:
Hmm, given your other comment, I'm still slightly confused.  I'm sure you can explain it to me.

kw['components'] is passed in from install.py, and there, components is set by detect_components(), which properly chooses default_components.rpm or .deb, depending on host.  So I fully understand why kw.pop('components') should get the right thing...

But, doesn't kw.pop('components', default_components) assign packages to default_components if kw['components'] does not exist?  And additionally, if it does exist, but evaluates to False, the "or default_components" bit also sets packages = default_components.

And at the top, we have "from ceph_deploy.util.constants import default_components", so isn't that going to just import the NamedTuple?  When using "default_components" directly like this, don't we still have to designing .rpm or .deb?

I may not understand how NamedTuples work.  Will check on that.  :)